encryption.md

```diff
@@ -0,0 +1,52 @@
+## Message Encryption
+
+DIDComm supports two types of message encryption: Authenticated Sender Encryption and Anonymous Sender Encryption. Both forms are encrypted to the recipient, but only Authenticated Sender Encryption provides assurances of who the sender is. 
+
+The encrypted form of a JWM is a JWE. The JOSE family defines [JSON Web Algorithms](https://tools.ietf.org/html/rfc7518) (JWAs) which standardize certain cryptographic operations that are related to preparing JOSE structures. For the purposes of interoperability, DIDComm messaging does not support all JWAs, rather it takes a subset of the supported algorithms that are applicable for the following cases around secure messaging. These supported algorithms are listed here.
+
+
+### Sender Authenticated Encryption
+
+When a sender would like to encrypt a message for a recipient or multiple recipients and also be authenticated such that the recipients can determine they were the party that encrypted the message, the JWA of `ECDH-1PU` defined by [draft](https://tools.ietf.org/html/draft-madden-jose-ecdh-1pu-02) SHOULD be used within the structure of a JWE. 
+
+For the keys involved in key agreement, the following elliptic curves MUST be supported.
+
+| Curve  | Description                                                  |
+| ------ | ------------------------------------------------------------ |
+| X25519 | The underlying curve is actually `Curve25519`, however when used in the context of Diffie-Hellman the identifier of `X25519` is used |
+| P-256  | NIST defined P-256 elliptic curve                            |
```

Comment:
Although `P-256` curve is the correct name, JWA also specifies the key wrapping algorithm using this curve. It's better to mention the algorithm value, especially that we're talking about ECDH here.

Since we're talking about ECDH-1PU encryption here, the kw algorithm would be: `ECDH-1PU+A256KW`. It's not part of the JWA spec yet, but it's defined in the ECDH-1PU draft link mentioned above this table.

Reply:
Probably adding a key agreement `algorithm` table will clarify this.

Reply:
added